src/reflection.jl

```diff
@@ -35,53 +36,63 @@ function transform(::Val{:CuFunction}, callsite, callexpr, CI, mi, slottypes; pa
     return Callsite(callsite.id, CuCallInfo(callinfo(Tuple{widenconst(ft), tt.val.parameters...}, Nothing, params=params)), callsite.head)
 end
 
-function find_callsites(CI::Core.CodeInfo, mi::Core.MethodInstance, slottypes; params=current_params(), multichoose::Bool=false, kwargs...)
+function find_callsites(CI::Core.CodeInfo, stmt_info::Union{Vector, Nothing}, mi::Core.MethodInstance, slottypes; params=current_params(), multichoose::Bool=false, kwargs...)
     sptypes = sptypes_from_meth_instance(mi)
     callsites = Callsite[]
 
-    function process_return_type(id, c, rt)
-        callinfo = nothing
-        is_call = isexpr(c, :call)
-        arg_base = is_call ? 0 : 1
-        length(c.args) == (arg_base + 3) || return nothing
-        ft = argextype(c.args[arg_base + 2], CI, sptypes, slottypes)
-        if isa(ft, Const)
-            ft = ft.val
-        end
-        if isa(ft, Function)
-            ft = typeof(ft)
-        end
-        argTs = argextype(c.args[arg_base + 3], CI, sptypes, slottypes)
-        if isa(argTs, Const)
-            sig = Tuple{widenconst(ft), argTs.val.parameters...}
-            miinner = multichoose ? choose_method_instance(sig) : first_method_instance(sig)
-            if miinner !== nothing
-                callinfo = MICallInfo(miinner, rt.val)
-            else
-                callinfo = FailedCallInfo(sig, rt)
+    for (id, c) in enumerate(CI.code)
+        callsite = nothing
+        if stmt_info !== nothing
+            info = stmt_info[id]
+            if info !== nothing
+                rt = CI.ssavaluetypes[id]
+                was_return_type = false
+                if isa(info, Core.Compiler.ReturnTypeCallInfo)
+                    info = info.info
+                    was_return_type = true
+                end
+                if isa(info, MethodMatchInfo)
+                    if info.results === missing
+                        continue
```

Comment:
In the early days should we `@warn`? We could suppress that before making a release, perhaps.

Reply:
This happens when there is too many applicable matches. That should also generate an entry in the inference msgs. Those aren't hooked up in the UI yet, but that's one of the next things I want to do.